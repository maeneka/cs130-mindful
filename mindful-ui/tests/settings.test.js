import React from 'react'
import Settings from '../src/components/settings';
import { h } from 'preact';
import renderer from 'react-test-renderer';
// See: https://github.com/preactjs/enzyme-adapter-preact-pure
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Test of the Settings UI', () => {
	test('Settings UI for all front end elements', () => {
		const wrapper = shallow(<Settings />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	test('Settings UI text displays', () => {
		const wrapper = shallow(<Settings />);
		expect(wrapper.find('h3').text()).toBe('mindful settings');
	});

	test('Settings UI back to dashboard', () => {
		const wrapper = shallow(<Settings />);
		expect(wrapper.find('a').find('p').text()).toBe('DASHBOARD');
	});

});
