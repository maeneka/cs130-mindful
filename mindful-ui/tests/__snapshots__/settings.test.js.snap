// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of the Settings UI Settings UI for all front end elements 1`] = `
<div
  className="page"
>
  <h3>
    mindful settings
  </h3>
  <div
    className="entries"
  >
    <table
      className="table"
    >
      <tr>
        <th />
        <th>
          MY SITES
        </th>
        <th>
          HOURS
        </th>
      </tr>
      <Entry
        action="remove"
        dec={[Function]}
        inc={[Function]}
        rem={[Function]}
        time={2}
        url="www.facebook.com"
      />
      <Entry
        action="remove"
        dec={[Function]}
        inc={[Function]}
        rem={[Function]}
        time={1}
        url="www.instagram.com"
      />
      <Entry
        action="remove"
        dec={[Function]}
        inc={[Function]}
        rem={[Function]}
        time={1}
        url="www.twitter.com"
      />
      <tr>
        <td>
          <a
            className="btn"
            href="#"
            onClick={[Function]}
          >
            <img
              src="/assets/images/+.png"
            />
          </a>
        </td>
        <td>
          <input
            className="inpt"
            oninput={[Function]}
            value=""
          />
        </td>
        <td>
          <div
            className="right"
          >
            <a
              className="btn"
              href="#"
            >
              <img
                onClick={[Function]}
                src="/assets/images/<.png"
              />
            </a>
            1
            <a
              className="btn"
              href="#"
            >
              <img
                onClick={[Function]}
                src="/assets/images/>.png"
              />
            </a>
          </div>
        </td>
      </tr>
      <tr>
        <th />
        <th
          className="gray"
        >
          MINDFUL SUGGESTIONS
        </th>
        <th />
      </tr>
      <Entry
        action="add"
        add={[Function]}
        dec={[Function]}
        inc={[Function]}
        time={2}
        url="www.reddit.com"
      />
    </table>
  </div>
  <a
    className="small-button"
    href="#"
    onClick={[Function]}
  >
    <p>
      DASHBOARD
    </p>
  </a>
</div>
`;
